### DISCUSSION TITLE
Threading G76

### DISCUSSION THREAD

**Discussion by Reprosense** [grblHAL/core#386] [Q&A] [answered]
hi , im trying to use a g76 code , but it seems shows an error after im trial this a little bit , did i send the wrong code or should i activated some function before tested it ( im using nucle f446 re and add a pin for encoder in protoneer v3 ) , here i send the video

https://github.com/grblHAL/core/assets/135573322/1bbc21ff-81ea-405b-a607-f6f0e0515ff1

i

**Comment by terjeio** [comment]
What are your settings? E.g. the alarm could be due to $341 set to a too small value.
Have you tuned the PID parameters? Before attempting G76 you should try G33 to check that multiple passes line up correctly, do they? I color a rod with a marker pen and [run passes with the tool just touching it](https://www.youtube.com/watch?v=bWQacuiWIxc).
See the [Wiki](https://github.com/grblHAL/core/wiki/Spindle-sync). If you are using Windows I recommend ioSender to configure the controller.

**Reply by Reprosense** [reply, marked as answer]
Oh , i found it , it seems that my Index ( Z ) signal is broken , thats why its stuck on Run
